pkg/pod/pod.go

```diff
@@ -322,6 +327,39 @@ func (b *Builder) Build(ctx context.Context, taskRun *v1beta1.TaskRun, taskSpec
 		return nil, err
 	}
 
+	readonly := true
+	if config.IsSpireEnabled(ctx) {
+		// add SPIRE's CSI volume to the explicitly declared use volumes
+		volumes = append(volumes, corev1.Volume{
+			Name: spire.WorkloadAPI,
+			VolumeSource: corev1.VolumeSource{
+				CSI: &corev1.CSIVolumeSource{
```

Comment:
CSI drivers require use of [privileged daemonsets](https://github.com/spiffe/spiffe-csi/blob/2aae3aaf6fa7afd6656bb0c26771d380452f515b/example/config/spiffe-csi-driver.yaml#L64) which makes isolation guarantees murky and ironically may prevent us from guaranteeing SLSA L3.

I'd rather use the [k8s_psat attestor](https://github.com/spiffe/spire/blob/v1.6.3/doc/plugin_agent_nodeattestor_k8s_psat.md) uses the service account oidc token to generate SVID instead, which IIUC won't require privilege. We can generate a local socket in the entrypoint for caching if we need to.

Reply:
@wlynch Thanks for reviewing. 
We do use the k8s_psat attestor for node attestation.

For the Workload API, the workload api ideally should also be supported over tcp also (https://github.com/spiffe/spiffe/blob/main/standards/SPIFFE_Workload_Endpoint.md#4-locating-the-endpoint) but seems like SPIRE agent only supports it over UDS. (The socket_path in agent configuration at https://spiffe.io/docs/latest/deploying/spire_agent/ ).
Without the CSI volume we would have to create a hostpath on every workload, which is supposedly a riskier option. The CSI volume plugin handles this. https://github.com/spiffe/spiffe-csi.

Reply:
> CSI drivers require use of [privileged daemonsets](https://github.com/spiffe/spiffe-csi/blob/2aae3aaf6fa7afd6656bb0c26771d380452f515b/example/config/spiffe-csi-driver.yaml#L64) which makes isolation guarantees murky and ironically may prevent us from guaranteeing SLSA L3.

@wlynch  Could you explain why isolation guarantees would be impacted with the CSI driver. My naive understanding is that we should be trust the spiffee csi daemonset the same way we'd trust any other the controller or any other service that is part of the build system

Reply:
(Note there is an issue capturing this concern as well at https://github.com/tektoncd/pipeline/issues/6559)

Reply:
Resolved on https://github.com/tektoncd/pipeline/issues/6559 🙂